### PR TITLE
Add generic request method. #258

### PR DESCRIPTION
**Pull request by tokuhirom**
SDK users can call APIs without waiting for SDK's development(usually, SDK's development is a bit delayed from API release)

### CODE REVIEW

**Comment by tokuhirom** [comment]
If the committers have a better idea to implement this, especially the interface, please suggest it :)

**Review by kkdai — APPROVED** [review]
LGTM.
@clsung do you have any suggestions?

**Comment by clsung** [comment]
The assumption is based on calling API in raw, how about following [net/http](https://golang.org/pkg/net/http/), i.e.

`func NewRawCall(method, url string, body io.Reader) ...`

Then we don't need to spawn different RawCalls and add PatchRawCall in the future?

**Comment by tokuhirom** [comment]
@kkdai @clsung 
Thanks for your review. I totally agreed.
I have combined multiple methods into one.
Could you review the interface again?

File: linebot/client.go [hunk 1]
```diff
@@ -185,6 +185,10 @@ func (client *Client) do(ctx context.Context, req *http.Request) (*http.Response
 	return client.httpClient.Do(req)
 }
 
+func (client *Client) request(ctx context.Context, req *http.Request) (*http.Response, error) {
```
Comment (clsung) [inline linebot/client.go:188]: I think we can just use
`call.c.do()`

File: linebot/raw.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+// WithContext method
+func (call *RawCall) WithContext(ctx context.Context) *RawCall {
+	call.ctx = ctx
+	return call
+}
+
+// Do method. Callee must close response object.
+func (call *RawCall) Do() (*http.Response, error) {
+	return call.c.request(call.ctx, call.req)
```
Comment (clsung) [inline linebot/raw.go:68]: ditto

File: linebot/client.go (hunk 1 above)
Comment (tokuhirom) [reply linebot/client.go]: Ah.. yep. I removed this method.

File: linebot/raw.go (hunk 2 above)
Comment (tokuhirom) [reply linebot/raw.go]: I changed code to use `call.c.do` directly

**Review by clsung — APPROVED** [review]
LGTM

**Pull request merged**